### PR TITLE
Preserve EventMesh precomputed hashes

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Preserve EventMesh precomputed hashes.

**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/17951

### CODE REVIEW

File: components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go [hunk 1]
```diff
@@ -563,6 +563,110 @@ func TestReconciler_APIRuleConfig_Upgrade(t *testing.T) {
...
+				)
+			}(),
+			givenReconcilerSetup: func(s *eventingv1alpha2.Subscription) (*Reconciler, client.Client) {
+				te := setupTestEnvironment(t, s)
+				te.backend.On("Initialize", mock.Anything).Return(nil)
+				te.backend.On("SyncSubscription", mock.Anything, mock.Anything, mock.Anything).Return(true, nil)
+				return NewReconciler(ctx, te.fakeClient, te.logger, te.recorder, te.cfg, te.cleaner,
+					te.backend, te.credentials, te.mapper, validator, collector), te.fakeClient
+			},
+			wantEv2Hash:            int64(2118518533334734626),
```
Comment (k15r) [inline components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go:631]: Can we extract constants for tests like these. Its really hard to read those random numbers and nobody should actually think (s)he has to compare/understand those values

Comment (marcobebway) [reply components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go]: Done

**Review by k15r — APPROVED** [review]

**Pull request merged**